frontend/src/app/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,105 @@
+import { User, onAuthStateChanged } from "firebase/auth";
+import React, { ReactNode, useContext, useEffect, useMemo, useState } from "react";
+
+import { auth } from "../../firebase/firebase";
+import { setCurrentUserInfo } from "../api/authHeaders";
+import { get } from "../api/requests";
+
+type UserReturn = {
+  id: string;
+  role: string;
+};
+
+type ApiUserResponse = {
+  _id: string;
+  role: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  address: Record<string, unknown>;
+  roleSpecificInfo: Record<string, unknown>;
+};
+
+const getUserIdAndRole = async (email: string): Promise<UserReturn> => {
```

Comment:
we should probably add some logic here that redirects back to the login page if the user exists in the firebase db, but not in the mongo db. Could be a separate issue/PR